python/core/auto_generated/qgsdataprovider.sip.in

```diff
@@ -116,7 +116,7 @@ Gets the data source specification.
 .. versionadded:: 3.0
 %End
 
-    virtual QgsDataProviderTemporalCapabilities *temporalCapabilities();
+    virtual QgsDataProviderTemporalCapabilities *temporalCapabilities() const;
```

Comment:
There's discussion about this here: https://github.com/qgis/QGIS/pull/35466#discussion_r401269274